authz/providers/azure/rbac/rbac.go

```diff
@@ -375,7 +378,94 @@ func (a *AccessInfo) CheckAccess(request *authzv1.SubjectAccessReviewSpec) (*aut
 
 		finalStatus = status
 	}
-	return finalStatus, nil
+	if finalStatus != nil && finalStatus.Allowed {
+		klog.V(5).Infof("Checkaccess request is allowed for user %s", checkAccessUsername)
+		return finalStatus, nil
+	}
+
+	if !a.enableManagedNamespaceRBAC || a.clusterType != managedClusters {
+		klog.V(5).Infof("Checkaccess request is denied for user %s", checkAccessUsername)
+		return finalStatus, nil
+	}
+
+	klog.V(5).Infof("Falling back to checking managed namespace scope", checkAccessUsername)
+
+	checkAccessURLManagedNS := *a.apiURL
+	checkAccessURLManagedNS.Path = path.Join(checkAccessURLManagedNS.Path, a.azureResourceId)
+
+	exists, managedNameSpaceString := getManagedNameSpaceScope(request)
+	if !exists {
+		// cluster level request
+		klog.V(5).Infof("Skipping managed namespace check for user %s because SAR is cluster scoped", checkAccessUsername)
+		return finalStatus, nil
+	}
+	checkAccessURLManagedNS.Path = path.Join(checkAccessURLManagedNS.Path, managedNameSpaceString)
+	checkAccessURLManagedNS.Path = path.Join(checkAccessURLManagedNS.Path, checkAccessPath)
+	paramsManagedNS := url.Values{}
+	paramsManagedNS.Add("api-version", checkAccessAPIVersion)
+	checkAccessURLManagedNS.RawQuery = paramsManagedNS.Encode()
+
+	ctx, cancel = context.WithTimeout(context.Background(), checkaccessContextTimeout)
+	defer cancel()
+	eg, egCtx = errgroup.WithContext(ctx)
+
+	ch = make(chan *authzv1.SubjectAccessReviewStatus, len(checkAccessBodies))
+	if len(checkAccessBodies) > 1 {
+		klog.V(5).Infof("Number of checkaccess requests to make: %d", len(checkAccessBodies))
+	}
+	eg.SetLimit(len(checkAccessBodies))
+	for _, checkAccessBody := range checkAccessBodies {
+		body := checkAccessBody
+		body.Resource.Id = path.Join(a.azureResourceId, managedNameSpaceString)
```

Comment:
Why we need this line?

Reply:
If the scope we are checking and the Resource.Id are mismatched, I get an error from ARM. That's why I had to update the Resource.Id